gdk4/src/event.rs

```diff
@@ -0,0 +1,616 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{
+    keys::Key, AxisUse, CrossingMode, Device, Display, Drop, EventType, KeyMatch, ModifierType,
+    NotifyType, ScrollDirection, Seat, Surface, TimeCoord, TouchpadGesturePhase,
+};
+use glib::translate::*;
+use glib::{StaticType, Type};
+use std::fmt;
+use std::mem;
+
+glib::wrapper! {
+    #[derive(Debug)]
+    pub struct Event(Shared<ffi::GdkEvent>);
+
+    match fn {
+        ref => |ptr| ffi::gdk_event_ref(ptr),
+        unref => |ptr| ffi::gdk_event_unref(ptr),
+    }
+}
+
+pub const NONE_EVENT: Option<&Event> = None;
+
+impl StaticType for Event {
+    #[doc(alias = "gdk_event_get_type")]
+    fn static_type() -> Type {
+        unsafe { from_glib(ffi::gdk_event_get_type()) }
+    }
+}
+
+impl Event {
+    pub fn downcast<T: EventTypeTrait>(self) -> Result<T, Event> {
+        unsafe {
+            if T::event_types().contains(&self.get_event_type()) {
+                Ok(from_glib_full(self.to_glib_full()))
```

Comment:
You could simply transmute here

Reply:
I guess this one is not as simple the other one as I keep getting
```
cannot transmute between types of different sizes, or dependently-sized types
source type: `event::Event` (64 bits)
target type: `T` (this type does not have a fixed size)
```
with
```
Ok(mem::transmute::<Event, T>(self))
```

Reply:
I think you'd need to make `T: Sized` for this

Reply:
But should be fine either way, just some additional atomic operations.

Reply:
> I think you'd need to make T: Sized for this

I have tried that one already but didn't help :/ 

I guess we can leave this one as is